components/org.wso2.carbon.identity.api.user.push.device.handler.v1/src/main/resources/push-auth.yaml

```diff
@@ -0,0 +1,780 @@
+          description: Unauthorized
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+        403:
+          description: Forbidden
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+        409:
+          description: Conflict
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+
+        500:
+          description: Server Error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+      tags:
+        - me
+
+    get:
+      description: |
+        This API is used to get a list of the registered devices.<br/>
+        <b>Permission required:</b>
+         * /permission/admin/login <br/>
+          <b>OAuth2 Scopes:</b>
+         * internal_identity_mgt_view
+      summary: |
+        Returns registered devices list of the user.
+      responses:
+        200:
+          description: All availabe devices of the user.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/AllDevices'
+        400:
+          description: Bad Request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+        401:
+          description: Unauthorized
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+        403:
+          description: Forbidden
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+        404:
+          description: Not found
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+        500:
+          description: Server Error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+
+      tags:
+        - me
+
+  /me/push-auth/devices/{deviceId}:
+    delete:
+      description: |
+        This API is used to remove a registered device via My Account.<br/>
+        <b>Permission required:</b>
+         * /permission/admin/login <br/>
+          <b>OAuth2 Scopes:</b>
+         * internal_identity_mgt_delete
+      summary: |
+        Remove a registered device.
+      parameters:
+        - name: deviceId
+          in: path
+          description: Unique Id of the device
+          required: true
+          schema:
+            type: string
+      responses:
+        200:
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/StatusDTO'
+        204:
+          description: No content
+        404:
```

Comment:
When a delete call for a non-exisiting deviceId is used do we get a 404?

Reply:
Removed